List<T>.cast<U> doesn't work.

**Describe the bug**
Compiling this code leads to an error regarding "lambda capture list".

This, however, doesn't seem to be the actual error as C++ compiler assumes that the generated **[object Object]** is a lambda capture list.

```js
namespace NSMain;

entrypoint function main(): List<Float64> {
    let inputs = List<Int>@{2, 5, 7, 3, 9};
    return inputs.cast<Float64>();
}
```

```shell
Error -- Error: Command failed: clang++  -g -DBDEBUG -std=c++17 -o neuralnet /workspaces/BosqueLanguage/impl/bin/scratch/cpp/*.cpp
/workspaces/BosqueLanguage/impl/bin/scratch/cpp/main.cpp:46:194: error: expected ',' or ']' in lambda capture list
    if(lI34I->entries.size() != 0) { BSQ_ABORT("Invalic element to cast", "/workspaces/BosqueLanguage/impl/bin/core/cpp/list.bsq", 43); } auto $$return = BSQ_NEW_ADD_SCOPE(_scope_I36I, [object Object], MIRNominalTypeEnum::nscore__list_t_nscore__float64_I10I);
                                                                                                                                                                                                 ^
1 error generated.
```

I have followed the compilation a bit and it seems that `generateTypeCheck` in cppbody_emitter.ts can't work with given `ctype`.

`const codetc = this.generateTypeCheck("v", ctype, ctype, rlctype);`

But the `idecl`, `ctype` and `ltype` variables contain the needed information for type construction, I think.

This is how their JSON representations look like.

```shell
{
   "enclosingDecl": "NSCore::List<T=NSCore::Int>",
   "name": "s_cast",
   "iname": "NSCore::List::s_cast",
   "key": "NSCore::List::s_cast<T=NSCore::Int, U=NSCore::Float64>",
   "sourceLocation": {
      "line": 43,
      "column": 0,
      "pos": 2831,
      "span": 6
   },
   "srcFile": "/workspaces/BosqueLanguage/impl/bin/core/cpp/list.bsq",
   "attributes": [
      "hidden",
      "recursive?"
   ],
   "recursive": false,
   "pragmas": [],
   "params": [
      {
         "name": "l",
         "type": "NSCore::List<T=NSCore::Int>"
      }
   ],
   "resultType": "NSCore::List<T=NSCore::Float64>",
   "implkey": "list_cast",
   "binds": {},
   "pcodes": {}
}
{
   "trkey": "NSCore::List<T=NSCore::Int>",
   "options": [
      {
         "trkey": "NSCore::List<T=NSCore::Int>",
         "ekey": "NSCore::List<T=NSCore::Int>"
      }
   ]
}
{
   "trkey": "NSCore::Int",
   "options": [
      {
         "trkey": "NSCore::Int",
         "ekey": "NSCore::Int"
      }
   ]
}
```

--- a/impl/src/core/cpp/core.bsq
+++ b/impl/src/core/cpp/core.bsq
@@ -132,6 +132,8 @@ struct entity Int provides KeyType, Parsable, PODType, APIType, Some {
     static stringify(i: Int): String # int_tostring
     static parse(str: String): Int # int_parse
 
+    method toFloat64(): Float64 # int_tofloat
+
     static max(v1: Int, v2: Int): Int {
         return v1 > v2 ? v1 : v2;
     }
@@ -172,6 +174,8 @@ struct entity Float64 provides Parsable, PODType, APIType, Some {
     static stringify(f: Float64): String # float64_tostring
     static parse(str: String): Float64 # float64_parse
 
+    method toInt(): Int # float_toint
+    
     method isInfinite(): Bool # float64_isinfinity
     method isNan(): Bool # float64_isnan
 

--- a/impl/src/core/symbolic/core.bsq
+++ b/impl/src/core/symbolic/core.bsq
@@ -132,6 +132,8 @@ struct entity Int provides KeyType, Parsable, PODType, APIType, Some {
     static stringify(i: Int): String # int_tostring
     static parse(str: String): Int # int_parse
 
+    method toFloat64(): Float64 # int_tofloat
+
     static max(v1: Int, v2: Int): Int {
         return v1 > v2 ? v1 : v2;
     }
@@ -177,6 +179,8 @@ struct entity Float64 provides Parsable, PODType, APIType, Some {
     static stringify(f: Float64): String # float64_tostring
     static parse(str: String): Float64 # float64_parse
 
+    method toInt(): Int # float_toint
+    
     method isInfinite(): Bool # float64_isinfinity
     method isNan(): Bool # float64_isnan
 

--- a/impl/src/core/verify/core.bsq
+++ b/impl/src/core/verify/core.bsq
@@ -132,6 +132,8 @@ struct entity Int provides KeyType, Parsable, PODType, APIType, Some {
     static stringify(i: Int): String # int_tostring
     static parse(str: String): Int # int_parse
 
+    method toFloat64(): Float64 # int_tofloat
+
     static max(v1: Int, v2: Int): Int {
         return v1 > v2 ? v1 : v2;
     }
@@ -171,6 +173,8 @@ struct entity Float64 provides Parsable, PODType, APIType, Some {
 
     static stringify(f: Float64): String # float64_tostring
     static parse(str: String): Float64 # float64_parse
+
+    method toInt(): Int # float_toint
 
     method isInfinite(): Bool # float64_isinfinity
     method isNan(): Bool # float64_isnan

--- a/impl/src/tooling/aot/cppbody_emitter.ts
+++ b/impl/src/tooling/aot/cppbody_emitter.ts
@@ -1830,7 +1830,14 @@ class CPPBodyEmitter {
                 bodystr = `auto $$return = BSQEnum{ (uint32_t)${params[0]}, MIRNominalTypeEnum::${this.typegen.mangleStringForCpp(this.currentRType.trkey)} };`;
                 break;
             }
-
+            case "int_tofloat": {
+                bodystr = `auto $$return = (double)${params[0]};`;
+                break;
+            }
+            case "int_mod": {
+                bodystr = `BSQ_ASSERT(0 < ${params[1]}, "Modulo must be a positive number"); auto $$return = ${params[0]} % ${params[1]};`;
+                break;
+            }
             case "float_min_value": {
                 bodystr = `auto $$return = std::numeric_limits<double>::min();`;
                 break;
@@ -1857,6 +1864,10 @@ class CPPBodyEmitter {
             }
             case "float64_parse": {
                 bodystr = `auto $$return = ;`;
+                break;
+            }
+            case "float_toint": {
+                bodystr = `BSQ_ASSERT(MIN_BSQ_INT <= ${params[0]} && ${params[0]} <= MAX_BSQ_INT, "Float64 is out of Int range"); auto $$return = (int64_t)${params[0]};`
                 break;
             }
             case "float64_isinfinity": {

